src/main/scala/firrtl/passes/wiring/WiringTransform.scala

```diff
@@ -36,9 +38,16 @@ case class SinkAnnotation(target: Named, pin: String) extends
   *
   * @throws WiringException if a sink is equidistant to two sources
   */
-class WiringTransform extends Transform {
-  def inputForm: CircuitForm = MidForm
-  def outputForm: CircuitForm = HighForm
+class WiringTransform extends Transform with DependencyAPIMigration {
+
+  override lazy val prerequisites = Forms.MidForm
+  override lazy val optionalPrerequisites = Seq.empty
+  override lazy val dependents = Forms.MidEmitters
+
+  override def invalidates(a: Transform): Boolean = {
+    val everything = new mutable.LinkedHashSet[Dependency[Transform]] ++ Forms.VerilogOptimized
+    (everything -- Forms.MinimalHighForm)(Dependency.fromTransform(a))
+  }
```

Comment:
```suggestion
    val myInvalidates = Forms.VerilogOptimized.toSet -- Forms.MinimalHighForm
    myInvalidates.contains(Dependency.fromTransform(a))
  }
```
Can you describe what this is doing a little? Also is there any point to using a `LinkedHashSet` since we're ultimately just doing a lookup?

Reply:
To follow on to this, the set gets re-created one every `invalidates` call.

It might be worth providing a trait that lets users express their invalidates as a set of `Dependency[Transform]`, since that seems like it will be a common pattern.

Reply:
I was thinking the same thing and actually toyed around in Scastie a little, you could define `myInvalidates` as a private field in the class. I think it just depends on how often it gets called. Is the memory cost of the `Set` worth keeping around for the full compilation? I'm thinking no if this function only gets called 1-2 times per object, but if it's several times it may become worth it.

Reply:
What this is saying is that wiring invalidates everything in the locally observable universe except `ChirrtlToHighFirrtl`.

Because this is `inputForm=MidForm`/`outputForm=HighForm` this means that this, per the original algorithm in `mergeTransforms`, needs to rerun `IRToWorkingIR`, `ResolveAndCheck`, and `HighFirrtlToMiddleFirrtl`. 

Currently, `firrtl.stage.Forms` describes lower forms as supersets of higher forms (`Forms.LowForm` is `Forms.MidForm` with additional transforms), so this PR constructs a bigger set and removes what it doesn't invalidate. This PR is the same as [this case](https://github.com/freechipsproject/firrtl/blob/master/src/main/scala/firrtl/Compiler.scala#L258).

Would it be useful to document this on the method?

Reply:
@albert-magyar: a trait to specify invalidates this way makes sense. The main motivation for allowing for function-based sets with invalidates as it seemed like a representation of the universal set was required (a transform should be able to invalidate everything).

The `invalidates` method will get called inside the `DependencyManager` everything this transform runs. It seems reasonable to go from `O(n)` object creations to `O(1)` object creations. (A `DependencyManager` only ever constructs one instance of a transform.)

I'll update this to use a `private val`.

Reply:
Note that `Set[A]` extends `A => Boolean` so I think this `Dependency[Transform]` options should be `Dependency[Transform] => Boolean` and then could be implemented as a `Set`. The only problem is that I don't think you can implement a method as a value function, you have to have a separate private val and do something like this:
```scala
trait Foo { def func(x: Int): Boolean }
class Bar extends Foo {
  private val funcImpl = Set(...)
  def func(x: Int): Boolean = funcImpl(x)
}
```

Reply:
Additional commit (with @jackkoenig attribution) added to include the suggested fix. I'll relegate an improvement to the way to specify invalidaties to future work.